server/bleep/src/indexes/file.rs

```diff
@@ -194,17 +171,33 @@ impl Indexer<File> {
         // hits is a mapping between a document address and the number of trigrams in it that
         // matched the query
         let repo_ref_term = Term::from_field_text(self.source.repo_ref, &repo_ref.to_string());
+        let branch_term = branch
+            .map(|b| {
+                trigrams(b)
```

Comment:
Would this match two branches which have the same set of trigrams in a different order?

Reply:
I've tried resolving this issue, but moving to a [whitespace tokenizer](https://github.com/BloopAI/bloop/commit/36722257b49570f0588135f5eabc4c6fca467879) triggers an infinite loop in tantivy that eats up all the memory. We've seen this bug in passing, but here it's reliable.

Will stick with this approach for now, and revisit this once we upgrade tantivy to 0.20+

Reply:
>Would this match two branches which have the same set of trigrams in a different order?

yes, fixable by turning branches to tokenize differently, or by applying a regex filter after the trigram matching work.